PythonVisualizations/SimpleQueue.py

```diff
@@ -6,357 +6,238 @@
     from .drawable import *
     from .VisualizationApp import *
 
-WIDTH = 800
-HEIGHT = 400
 
-CELL_SIZE = 50
-ARRAY_X0 = 100
-ARRAY_Y0 = 100
 
-class Queue(object):
-
-    colors = ['red', 'green', 'blue', 'orange', 'yellow', 'cyan', 'magenta',
-              'dodgerblue', 'turquoise', 'grey', 'gold', 'pink']
+class Queue(VisualizationApp):
+    WIDTH = 800
+    HEIGHT = 400
+    CELL_SIZE = 50
+    ARRAY_X0 = 100
+    ARRAY_Y0 = 100
     nextColor = 0
+    CELL_BORDER = 2
+    CELL_BORDER_COLOR = 'black'
 
-    def __init__(self, size=12):
+    def __init__(self, size=12, title="Queue", **kwargs):
+        super().__init__(title=title, **kwargs)
         self.list = [None]*size
         self.size = size
         self.front = 1  # when Queue is empty, front 
         self.rear = 0   # should be to right of rear.
         self.nItems = 0
-        
+        self.rearArrow = (None, None)
+        self.frontArrow = (None, None)
+        self.insertButton = None
+        self.deleteButton=None
+        self.display()
+
     def __str__(self):
         return str(self.list)
     
     # ANIMATION METHODS
-    def speed(self, sleepTime):
-        return (sleepTime * (scaleDefault + 50)) / (scale.get() + 50)
+    def createIndex(  # Create an index arrow to point at an indexed
+            self, index=-1, name=None, high=True):  # cell with an optional name label
+
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        x0 = cell_center[0]
+        x1 = x0
+        y0 = cell_coords[1] - self.CELL_SIZE * 2 // 2
+        y1 = cell_coords[1] - self.CELL_SIZE * 1 // 4
+        if high:  # changes height of arrow
+            y0 -= 20
+
+        arrow = self.canvas.create_line(
+            x0, y0, x1, y1, arrow="last", fill=self.VARIABLE_COLOR)
+        if name:
+            label = self.canvas.create_text(
+                x1 + 2, y0, text=name, anchor=SW,
+                font=self.VARIABLE_FONT, fill=self.VARIABLE_COLOR)
+        return (arrow, label) if name else (arrow,)
+
+    # def speed(self, sleepTime):
+    #     return (sleepTime * (scaleDefault + 50)) / (scale.get() + 50)
     
     def set(self, index, val):
         # reset the value of the Element at that index to val
         self.list[index].val = val
 
         # get the position of the displayed value
-        pos = canvas.coords(self.list[index].display_val)
+        pos = self.canvas.coords(self.list[index].display_val)
 
         # delete the displayed value and replace it with the updated value
-        canvas.delete(self.list[index].display_val)
-        self.list[index].display_val = canvas.create_text(pos[0], pos[1], text=str(val), font=('Helvetica', '20'))
+        self.canvas.delete(self.list[index].display_val)
+        self.list[index].display_val = self.canvas.create_text(pos[0], pos[1], text=str(val), font=('Helvetica', '20'))
 
         # update window
-        window.update()  
+        self.window.update()
```

Comment:
Defining a method with the same name as the Pyhton built-in data type, `set`, could be confusing.  The method is not being called anywhere that I can see, and neither is the built-in function.  Please rename this method or just get rid of it.